src/test/java/test/project/bookshop/repository/BookRepositoryTests.java

```diff
@@ -0,0 +1,77 @@
+package test.project.bookshop.repository;
+
+import java.util.Optional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import test.project.bookshop.model.Book;
+import test.project.bookshop.repository.book.BookRepository;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+public class BookRepositoryTests {
```

Comment:
How will you realize your application? Is it okay that it has some default books and categories?
Better add testing data only for tests and remove it after testing.